include/laser_filters/radius_outlier_filter.h

```diff
@@ -0,0 +1,190 @@
+
+#include <set>
+
+#include <filters/filter_base.h>
+#include <laser_filters/radius_outlier_detector.h>
+#include <laser_filters/RadiusOutlierFilterConfig.h>
+#include <dynamic_reconfigure/server.h>
+#include <sensor_msgs/LaserScan.h>
+#include <angles/angles.h>
+#include <math.h>
+
+
+namespace laser_filters
+{
+/** @b RadiusOutlierFilter filters scan points that don't have at least
+ *     some number of neighbors within a certain range.
+ */
+
+class RadiusOutlierFilter : public filters::FilterBase<sensor_msgs::LaserScan>
+{
+public:
+  double radius_search_;
+  int window_, min_neighbors_;
+
+  RadiusOutlierDetector radius_detector_;
+
+  std::shared_ptr<dynamic_reconfigure::Server<laser_filters::RadiusOutlierFilterConfig>> dyn_server_;
+  boost::recursive_mutex own_mutex_;
+  RadiusOutlierFilterConfig param_config;
+
+  ////////////////////////////////////////////////////////////////////////////////
+  RadiusOutlierFilter()
+  {
+  }
+
+  /**@b Configure the filter from XML */
+  bool configure()
+  {
+    ros::NodeHandle private_nh("~" + getName());
+    dyn_server_.reset(new dynamic_reconfigure::Server<laser_filters::RadiusOutlierFilterConfig>(own_mutex_, private_nh));
+    dynamic_reconfigure::Server<laser_filters::RadiusOutlierFilterConfig>::CallbackType f;
+    f = boost::bind(&laser_filters::RadiusOutlierFilter::reconfigureCB, this, _1, _2);
+    dyn_server_->setCallback(f);
+
+    if (!filters::FilterBase<sensor_msgs::LaserScan>::getParam(std::string("radius_search"), radius_search_))
+    {
+      ROS_ERROR("Error: RadiusOutlierFilter was not given radius_search.\n");
+      return false;
+    }
+    if (!filters::FilterBase<sensor_msgs::LaserScan>::getParam(std::string("min_neighbors"), min_neighbors_))
+    {
+      ROS_ERROR("Error: RadiusOutlierFilter was not given min_neighbors.\n");
+      return false;
+    }
+    if (!filters::FilterBase<sensor_msgs::LaserScan>::getParam(std::string("window"), window_))
+    {
+      ROS_ERROR("Error: RadiusOutlierFilter was not given window.\n");
+      return false;
+    }
+
+    if (radius_search_ < 0)
+    {
+      ROS_ERROR("radius_search must be >= 0. Forcing radius_search = 0.");
+      radius_search_ = 0.0;
+    }
+
+    if (min_neighbors_ < 0)
+    {
+      ROS_ERROR("min_neighbors must be >= 0. Forcing min_neighbors = 0");
+      min_neighbors_ = 0;
+    }
+    radius_detector_.configure(radius_search_);
+
+    param_config.radius_search = radius_search_;
+    param_config.window = window_;
+    param_config.min_neighbors = min_neighbors_;
+    dyn_server_->updateConfig(param_config);
+
+    return true;
+  }
+
+  void reconfigureCB(RadiusOutlierFilterConfig& config, uint32_t level)
+  {
+    radius_search_ = config.radius_search;
+    window_ = config.window;
+    min_neighbors_ = config.min_neighbors;
+
+    radius_detector_.configure(radius_search_);
+  }
+
+  ////////////////////////////////////////////////////////////////////////////////
```

Comment:
Could you remove the `/////////` lines from this file? I don't think they help with readability.

Reply:
Done in 89a46899b9dbf5643e36374263469c11b336f263.